### ISSUE TITLE
Clean up docs images and use matplotlib plot directive

### ISSUE THREAD

**Issue by albireox** [sdss/marvin#452]
We have a lot of static files in the documentation, most of which are plot snapshots, that are bloating the size of the repository (93MB for a newly cloned repo). In the future we should try to reorganise these and purge the ones that are not in use from the repo. Maybe the static images can be stored online, or we can create another repo for them. But, in general, we need to find a better way to store large files that doesn't involve committing them to the main repo.

We should also consider using the matplotlib [plot directive](https://matplotlib.org/devel/plot_directive.html) that allows to generate the plots dynamically when the documentation is built. The main problem is that RTD won't be able to access the data remotely since it requires authentication but maybe that will change once DR15 is public, or we can provide a subset of example data the same way seaborn or pandas do.

**Comment by albireox** [comment]
So, I actually implemented this and it works nicely. I added a `get_test_data` function to `conf.py` that downloads and uncompress a files to a local SAS if the documentation is being compiled at RTD. After that one can use marvin normally. For instance, this directive,

```rst
.. plot::
    :align: center

    import marvin
    import matplotlib.pyplot as plt

    my_cube = marvin.tools.Cube('7443-12703')
    flux = my_cube.flux
    spectrum = flux[:, 50, 55]
    ax = spectrum.plot(show_std=True)
    ax.set_xlim(6000, 8000)
```

works fine and produces the plot. 

![screenshot 2018-07-18 at 17 51 15](https://user-images.githubusercontent.com/568775/42915188-35bb666c-8ab3-11e8-813c-cda6eac548b4.png)

The only problem is where to put the data. For now I have stored the data in my dropbox and RTD downloads it absurdly quickly (800MB in 45 seconds). I tried storing it somewhere else (e.g., my UW FTP) and it's extremely slow. I assume they have a quick link to Dropbox. Right now the file contains 8485-1901, 7443-12701, and 7443-12703 (HYB10 only). We can add more.

Ultimately it would be nice if we can just use the API, i.e., have RTD set up a connection to the public data with a token and just use the remote.

**Comment by havok2063** [comment]
I think once we go public, RTD should be able to remotely access the public data.  Without tokens or any authentication.  As long as the dynamic plotting code only generates plots using DR data.   Maybe we can make an agreement with Dropbox to store SDSS data.   Otherwise, building our own RTD server at Utah is not a bad idea.  We've certainly talked about it for sdss5.   On an unrelated note, we could consider doing the same thing with Travis, or switching our testing suite to Jenkins, which requires you to build and host your own server.

**Comment by albireox** [comment]
Right, the DRs don't require tokens. I think that's what we should do once DR15 is out. I think our documentation (at least the most part) should refer to DRs since that's what most people will see. Dropbox is a patch for now but seems to work.

And yes, we should talk about moving some of the CI tools to Utah. RTDs may be the easiest but it's also the one that works best on its own site. If we can use their API to build our own public facing page for the documentation that may be enough.

Testing in the end probably will need to go to Utah, I think. Unless we split marvin into several components (tools, api, web) and set up a different testing suit for each one so that we take down the full test time.
